userapi/src/main/java/fk/prof/userapi/StorageFactory.java

```diff
@@ -25,27 +26,28 @@
 public class StorageFactory {
 
   private static final String S3 = "s3";
-    private static final String ACCESS_KEY = "access.key";
-    private static final String SECRET_KEY = "secret.key";
-    private static final String END_POINT = "endpoint";
+  private static final String ACCESS_KEY = "access.key";
+  private static final String SECRET_KEY = "secret.key";
+  private static final String END_POINT = "endpoint";
+  private static final String LIST_OBJECTS_TIMEOUT_MS = "list.objects.timeout.ms";
 
-    private static S3AsyncStorage s3AsyncStorage = null;
+  private static S3AsyncStorage s3AsyncStorage = null;
 
   synchronized public static AsyncStorage getAsyncStorage(JsonObject storageConfig, ExecutorService storageExecSvc) {
-    if (storageConfig.fieldNames().isEmpty())
-      return getS3AsyncInstance(storageConfig.getJsonObject(S3), storageExecSvc);
-    switch (storageConfig.fieldNames().iterator().next()) {
-            case S3:
-              return getS3AsyncInstance(storageConfig.getJsonObject(S3), storageExecSvc);
-            default:
-              return getS3AsyncInstance(storageConfig.getJsonObject(S3), storageExecSvc);
-        }
+    for (String key : storageConfig.fieldNames()) {
```

Comment:
Maybe not required? Since this class has an explicit dependency on s3 (it has member: S3AsyncStorage and not AsyncStorage), we can directly expect s3 storage config to be provided.
Other way to look at it is, to code this class to interface (AsyncStorage) in which case below switch makes sense